File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/clips/FollowedClipsFragment.java
package com.exact.twitch.ui.clips;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.exact.twitch.R;
import com.exact.twitch.ui.main.MainActivity;
import com.exact.twitch.ui.fragment.RadioButtonBottomSheetDialogFragment;
import com.exact.twitch.util.FragmentUtils;

public class FollowedClipsFragment extends BaseClipsFragment implements RadioButtonBottomSheetDialogFragment.OnOptionSelectedListener {

    private int[] sortOptions = new int[] { R.string.trending, R.string.view_count };
    private String userToken;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (isFragmentVisible()) {
            int defaultSortItem = sortOptions[1];
            if (getViewModel().getSelectedSortItem() == null) {
                getViewModel().setSelectedSortItem(defaultSortItem);
            }
            if (getViewModel().getSortText().getValue() == null) {
                getViewModel().getSortText().setValue(getString(defaultSortItem));
            }
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        userToken = requireActivity().getSharedPreferences("auth", Context.MODE_PRIVATE).getString("token", null);
        sortByContainer.setOnClickListener(v -> FragmentUtils.INSTANCE.showRadioButtonDialogFragment(
                requireActivity(),
                getChildFragmentManager(),
                sortOptions,
                getViewModel().getSelectedSortItem())
        );
        TextView textView = view.findViewById(R.id.fragment_clips_tv_login);
        Button button = view.findViewById(R.id.fragment_clips_btn_login);
        if (userToken == null) {
            textView.setVisibility(View.VISIBLE);
            button.setVisibility(View.VISIBLE);
            button.setOnClickListener(v -> ((MainActivity) getContext()).onLoginClicked());
            progressBar.setVisibility(View.GONE);
            sortByContainer.setVisibility(View.GONE);
        }
    }

    @Override
    public void onSelect(int optionId, CharSequence optionText) {
        if (getViewModel().getSelectedSortItem() != optionId) {
            getViewModel().setTrending(optionId == R.string.trending);
            getViewModel().setSelectedSortItem(optionId);
            getViewModel().getSortText().postValue(optionText);
            loadData(true);
        }
    }

    @Override
    public void loadData(boolean override) {
        getViewModel().loadFollowedClips(userToken, getViewModel().getTrending(), override);
    }

    @Override
    protected void initData() {
        loadData(false);
    }
}

File: app/src/main/java/com/exact/twitch/ui/clips/FollowedClipsFragment.kt
package com.exact.twitch.ui.clips

import android.content.Context
import android.os.Bundle
import android.view.View
import com.exact.twitch.R
import com.exact.twitch.model.User
import com.exact.twitch.util.C
import com.exact.twitch.util.FragmentUtils
import com.exact.twitch.util.TwitchApiHelper
import kotlinx.android.synthetic.main.fragment_clips.*

class FollowedClipsFragment : BaseClipsFragment() {

    private companion object {
        val sortOptions = listOf(R.string.trending, R.string.view_count)
        const val DEFAULT_INDEX = 1
        const val TAG = "FollowedClips"
    }

    private lateinit var user: User

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        user = arguments!!.getParcelable(C.USER)!!
        sortByRl.setOnClickListener { FragmentUtils.showRadioButtonDialogFragment(requireActivity(), childFragmentManager, sortOptions, DEFAULT_INDEX, TAG) }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if (isFragmentVisible && !viewModel.isInitialized()) {
            viewModel.sortText.postValue(getString(sortOptions[DEFAULT_INDEX]))
        }
    }

    override fun loadData(override: Boolean) {
        viewModel.loadFollowedClips(user.token, override)
    }

    override fun onSelect(index: Int, text: CharSequence, tag: Int?) {
        if (viewModel.sortText.value != text) {
            viewModel.trending = tag == R.string.trending
            viewModel.sortText.postValue(text)
            loadData(true)
        }
    }
}
